Makefile

```diff
@@ -1,13 +0,0 @@
-.PHONY: all
```

Comment:
I would like to keep the simple root Makefile with these two targets (all and clean) to preserve as much as possible the simplicity of being able to just clone and `make`.

I realise we don't want to maintain Makefile and CMake build systems in parallel but at least something simple that would do as you suggest - `mkdir build && cd build && cmake . && make` and clean up after